### ISSUE TITLE
opp_neddoc does not work in headless mode

### ISSUE THREAD

**Issue by haug-den-lucas** [omnetpp/omnetpp#1043]
I tried to set up a Gitlab pipeline which automatically generates my NED documentation.

`opp_neddoc` however, does not seem to work without a connected display (which seems weird, as it is supposed to be a command line tool)

What I've tried:

1: Use the omnetpp docker container:
Docker Container: ghcr.io/omnetpp/omnetpp:u22.04-6.0.1
Running opp_neddoc fails with "The IDE is not installed, but it is required by opp_neddoc!"

2. Use the omnetpp-gui docker container:
Docker Container: ghcr.io/omnetpp/omnetpp-gui:u22.05-6.0.1
Running opp_neddoc fails with
```
root@dc19269fad3d:~/inet# opp_neddoc .
OMNeT++ IDE:
An error has occurred. See the log file
/tmp/tmp.uttWk716CT/.metadata/.log.
```

So i changed 
https://github.com/omnetpp/omnetpp/blob/350e6791cad47c884ca08d9738c1340704443dc5/src/utils/opp_neddoc#L48
to `$LAUNCHER "$@" $WORKSPACE_ARGS 2>&1` to actually see some logs and then it fails with
```
root@dc19269fad3d:~/inet# opp_neddoc .
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
OMNeT++ IDE: Cannot open display: 
OMNeT++ IDE:
An error has occurred. See the log file
/tmp/tmp.Zn1RwJi5nq/.metadata/.log.
```

So the problem seems to be that opp_neddoc is only able to run on a machine with a display which feels quite weird as it is supposed to be a command line tool.
Is there any way to work around this?

**Comment by avarga** [comment]
Running `opp_neddoc` with `xvfb-run` seems to be the solution. It creates a virtual framebuffer for the duration of running the program you give it as argument.

Example: `xvfb-run opp_neddoc .`

**Comment by avarga** [comment]
> when installing xvfb inside the current omnetpp-gui docker container it seems to freeze 

That was unexpected, thanks for the info.

> So it would still be preferable if opp_neddoc could run natively on a headless machine.

We'll try to fix it in the IDE Java code, but we'll be unable to pull it off if it requires changes in the Eclipse platform libs such as SWT...

**Comment by rhornig** [comment]
Running with `xvfb-run` works correctly for me in a desktop shell:
```
GDK_BACKEND=x11 DISPLAY= xvfb-run opp_neddoc --verbose .
Generating NED documentation for 'aloha'.
............
```

I have not tried in a docker image though.

**Comment by haug-den-lucas** [comment]
> Running with `xvfb-run` works correctly for me in a desktop shell:
> 
> ```
> GDK_BACKEND=x11 DISPLAY= xvfb-run opp_neddoc --verbose .
> Generating NED documentation for 'aloha'.
> ............
> ```
> 
> I have not tried in a docker container though.


Yes, interestingly this works perfectly well for me even on a headless server:

```
X@X:~/tmp/inet4.5$ GDK_BACKEND=x11 DISPLAY= xvfb-run opp_neddoc --verbose .
Generating NED documentation for 'inet'.
..........^C
X@X:~/tmp/inet4.5$
```

However if i run the official omnetpp-gui docker container:


And execute the following script in it:
```
apt-get update
apt-get -y install xvfb
git clone --depth=1 https://github.com/inet-framework/inet.git
cd inet
```
and then
```
omnetpp-gui-6.0.1:/root/models/inet$ GDK_BACKEND=x11 DISPLAY= xvfb-run opp_neddoc --verbose .
Generating NED documentation for 'inet'.

^C
omnetpp-gui-6.0.1:/root/models/inet$
```
nothing happens until cancelling (I waited for multiple minutes, on the same machine outside the docker container the first dots appeared after a few seconds).

Interestingly in the second case there is an additional new line before canceling.

Or summarized: `xvfb-run opp_neddoc` works perfectly on my desktop, on a headless server but not in a docker container.
